src/app_service/service/ens/service.nim

```diff
@@ -217,20 +267,17 @@ QtObject:
       vptr: cast[ByteAddress](self.vptr),
       slot: "onEnsUsernameDetailsFetched",
       ensUsername: username,
-      chainId: self.networkService.getNetworkForEns().chainId,
+      chainId: chainId,
       isStatus: isStatus
     )
     self.threadpool.start(arg)
 
   proc extractCoordinates(self: Service, pubkey: string):tuple[x: string, y:string] =
     result = ("0x" & pubkey[4..67], "0x" & pubkey[68..131])
 
-  proc setPubKeyGasEstimate*(self: Service, ensUsername: string, address: string): int = 
+  proc setPubKeyGasEstimate*(self: Service, chainId: int, ensUsername: string, address: string): int = 
     try:
-      let
-        chainId = self.networkService.getNetworkForEns().chainId
```

Comment:
That looks more logical to not allow the service to not be called by any chain id